src/toil/batchSystems/gridengine.py

```diff
@@ -255,14 +260,20 @@ def getRunningBatchJobIDs(self):
         return self.worker.getRunningJobIDs()
     
     def getUpdatedBatchJob(self, maxWait):
-        i = self.getFromQueueSafely(self.updatedJobsQueue, maxWait)
+        i = self.updatedJobsQueue.get()
         if i == None:
             return None
         jobID, retcode = i
         self.updatedJobsQueue.task_done()
         self.currentjobs.remove(jobID)
         return i
-    
+
+    def shutdown(self):
+        """
+        Because the gridEngine Worker is a daemon thread this fn is unnecessary
```

Comment:
It shouldn't be a daemon thread. We need shutdown() to work because we start and stop a batch system during the unit tests. I hope the same techniques we used in the single-machine batch system will work here.